sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/StoreResultDiagnostics.java

```diff
@@ -0,0 +1,195 @@
+        private static final long serialVersionUID = 5315472126043077905L;
+
+        public StoreResultDiagnosticsSerializer(){
+            super(StoreResultDiagnostics.class);
+        }
+
+        @Override
+        public void serialize(StoreResultDiagnostics storeResultDiagnostics,
+                              JsonGenerator jsonGenerator,
+                              SerializerProvider serializerProvider) throws IOException {
+            StoreResponseDiagnostics storeResponseDiagnostics = storeResultDiagnostics.getStoreResponseDiagnostics();
+            jsonGenerator.writeStartObject();
+            jsonGenerator.writeObjectField("storePhysicalAddress", storeResultDiagnostics.storePhysicalAddressAsString);
+            jsonGenerator.writeNumberField("lsn", storeResultDiagnostics.lsn);
+            jsonGenerator.writeNumberField("globalCommittedLsn", storeResultDiagnostics.globalCommittedLSN);
+            jsonGenerator.writeStringField("partitionKeyRangeId", storeResponseDiagnostics.getPartitionKeyRangeId());
+            jsonGenerator.writeBooleanField("isValid", storeResultDiagnostics.isValid);
+            jsonGenerator.writeNumberField("statusCode", storeResponseDiagnostics.getStatusCode());
+            jsonGenerator.writeNumberField("subStatusCode", storeResponseDiagnostics.getSubStatusCode());
+            jsonGenerator.writeBooleanField("isGone", storeResultDiagnostics.isGoneException);
+            jsonGenerator.writeBooleanField("isNotFound", storeResultDiagnostics.isNotFoundException);
+            jsonGenerator.writeBooleanField("isInvalidPartition", storeResultDiagnostics.isInvalidPartitionException);
+            jsonGenerator.writeBooleanField("isThroughputControlRequestRateTooLarge", storeResultDiagnostics.isThroughputControlRequestRateTooLargeException);
+            jsonGenerator.writeNumberField("requestCharge", storeResponseDiagnostics.getRequestCharge());
+            jsonGenerator.writeNumberField("itemLSN", storeResultDiagnostics.itemLSN);
+            jsonGenerator.writeStringField("sessionToken", storeResponseDiagnostics.getSessionTokenAsString());
+            jsonGenerator.writeObjectField("backendLatencyInMs", storeResultDiagnostics.backendLatencyInMs);
+            this.writeNonNullObjectField(jsonGenerator, "exceptionMessage", storeResponseDiagnostics.getExceptionMessage());
+            this.writeNonNullObjectField(jsonGenerator, "exceptionResponseHeaders", storeResponseDiagnostics.getExceptionResponseHeaders());
+            jsonGenerator.writeObjectField("transportRequestTimeline", storeResponseDiagnostics.getRequestTimeline());
+
+            this.writeNonNullObjectField(jsonGenerator,"transportRequestChannelAcquisitionContext", storeResponseDiagnostics.getChannelAcquisitionTimeline());
+
+            jsonGenerator.writeObjectField("rntbdRequestLengthInBytes", storeResponseDiagnostics.getRntbdRequestLength());
```

Comment:
same question here: should we use writeNumberField?

Reply:
I don't want to change any existing fields, the way they were being written, but will make sure to fix the newly added string fields.